src/server/enrollment-system.js

```diff
@@ -904,10 +904,13 @@ function veteranToFinancialsInfo(veteran) {
   });
 
   const dependentFinancials = veteranToDependentFinancialsCollection(veteran);
-  const hasDependentFinancials = _.compact(dependentFinancials.dependentFinancials.map((child) => { return child.income; })).length > 0;
+  let hasDependentFinancials = false;
+  if (dependentFinancials) {
+    hasDependentFinancials = _.compact(dependentFinancials.dependentFinancials.map((child) => { return child.incomes; })).length > 0;
+  }
   const spouseFinancials = veteranToSpouseFinancials(veteran);
 
-  const hasIncomeData = expenses || incomes || spouseFinancials.incomes || hasDependentFinancials;
+  const hasIncomeData = expenses || incomes || spouseFinancials.spouseFinancials.incomes || hasDependentFinancials;
```

Comment:
Sorry, I'm having to rebase because I branched from a WIP branch that wasn't up to date.